### PR TITLE
Enable Add encryption

### PR DESCRIPTION
**Pull request by Eabramov84**

### CODE REVIEW

File: recipes-core/systemd/systemd_%.bbappend [hunk 1]
```diff
@@ -3,6 +3,10 @@ quotaon_srv = "quotaon.service"
 
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
+PACKAGECONFIG[p11kit] = "-Dp11kit=true -Dhomed=false, -Dp11kit=false, p11-kit"
```
Comment (al1img) [inline recipes-core/systemd/systemd_%.bbappend:15]: -Dp11kit=true, -Dp11kit=false is this correct options?

File: recipes-core/systemd/systemd-boot_%.bbappend [hunk 2]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (al1img) [inline recipes-core/systemd/systemd-boot_%.bbappend:1]: Remove empty line

File: recipes-support/lvm2/lvm2_%.bbappend [hunk 3]
```diff
@@ -0,0 +1,3 @@
+FILES_libdevmapper += " \
```
Comment (al1img) [inline recipes-support/lvm2/lvm2_%.bbappend:1]: Why do we need this changes?

File: recipes-security/softhsm/files/softhsm.module [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module: /usr/lib/softhsm/libsofthsm2.so
+managed: yes
```
Comment (al1img) [inline recipes-security/softhsm/files/softhsm.module:2]: Add new line at the end.

File: recipes-security/softhsm/softhsm_%.bbappend [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+
+SRC_URI += "\
+    file://softhsm.module \
+"
+
+do_install_append () {
+    install -d ${D}${sysconfdir}/pkcs11/modules/
+    install -m 0644 ${WORKDIR}/softhsm.module ${D}${sysconfdir}/pkcs11/modules/
+}
```
Comment (al1img) [inline recipes-security/softhsm/softhsm_%.bbappend:10]: Add new line at the end.

File: recipes-core/systemd/files/crypttab [hunk 6]
```diff
@@ -0,0 +1 @@
+aos_partition /dev/hda6 - pkcs11-uri=auto,nofail,noauto
```
Comment (al1img) [inline recipes-core/systemd/files/crypttab:1]: New line at the end.

File: recipes-aos/aos-iamanager/files/aos_iamanager.cfg [hunk 7]
```diff
@@ -63,6 +63,17 @@
             "Params": {
                 "StoragePath": "/var/aos/crypt/um/"
             }
+        },
+        {
+            "ID": "decrypt",
```
Comment (al1img) [inline recipes-aos/aos-iamanager/files/aos_iamanager.cfg]: diskencryption?

File: recipes-core/systemd/files/override.conf [hunk 8]
```diff
@@ -0,0 +1,6 @@
+[Service]
+Environment="PIN=1234"
```
Comment (rshym) [inline recipes-core/systemd/files/override.conf:2]: This looks like violation of corporate policies and may trigger scanners.

File: recipes-aos/aos-iamanager/files/aos_iamanager.cfg [hunk 9]
```diff
@@ -63,6 +63,17 @@
...
+            "Plugin": "pkcs11module",
+            "Algorithm": "rsa",
+            "MaxItems": 1,
+            "Disabled": false,
+            "Params": {
+                "library": "/usr/lib/softhsm/libsofthsm2.so",
+                "userPin": "1234"
```
Comment (rshym) [inline recipes-aos/aos-iamanager/files/aos_iamanager.cfg:79]: This looks like violation of corporate policies.

File: recipes-aos/aos-iamanager/files/encrypt.sh [hunk 10]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+echo "hello" | cryptsetup  -q --type luks2 luksFormat /dev/hda6
+
+PASSWORD=hello  PIN=1234 systemd-cryptenroll --pkcs11-token-uri="pkcs11:token=aos;object=diskencryption" /dev/hda6
```
Comment (rshym) [inline recipes-aos/aos-iamanager/files/encrypt.sh:5]: Looks like violation of policies.

File: recipes-aos/aos-iamanager/files/encrypt.sh [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+echo "hello" | cryptsetup luksRemoveKey /dev/hda6
+
+PIN=1234 /lib/systemd/systemd-cryptsetup attach aos_partition /dev/hda6 - pkcs11-uri=auto
```
Comment (rshym) [inline recipes-aos/aos-iamanager/files/encrypt.sh:9]: Looks like violation of policies.

File: recipes-core/systemd/systemd_%.bbappend (hunk 1 above)
Comment (Eabramov84) [reply recipes-core/systemd/systemd_%.bbappend:15]: yes, it is correct  plese see prameter description from yoicto manual
 PACKAGECONFIG[f1] = "--with-f1,--without-f1,build-deps-f1,rt-deps-f1"

File: recipes-core/systemd/systemd-boot_%.bbappend (hunk 2 above)
Comment (Eabramov84) [reply recipes-core/systemd/systemd-boot_%.bbappend:1]: removed

File: recipes-support/lvm2/lvm2_%.bbappend (hunk 3 above)
Comment (Eabramov84) [reply recipes-support/lvm2/lvm2_%.bbappend:1]: By default libdevmapper doesn't add udev rules to the system. These udev rules are necessary for systems crypto tools synchronisation.

File: recipes-security/softhsm/files/softhsm.module (hunk 4 above)
Comment (Eabramov84) [reply recipes-security/softhsm/files/softhsm.module:2]: added

File: recipes-core/systemd/files/crypttab (hunk 6 above)
Comment (Eabramov84) [reply recipes-core/systemd/files/crypttab:1]: added

File: recipes-aos/aos-iamanager/files/aos_iamanager.cfg (hunk 7 above)
Comment (Eabramov84) [reply recipes-aos/aos-iamanager/files/aos_iamanager.cfg]: fixed.

File: recipes-aos/aos-iamanager/files/encrypt.sh (hunk 11 above)
Comment (Eabramov84) [reply recipes-aos/aos-iamanager/files/encrypt.sh:9]: this implementation is only for a virtual machines for testing porpuses

**Review by al1img — APPROVED** [review]

**Comment by rshym** [comment]
May be I'm wrong, but please check that these two files are different and this is not misprint:
`"userPinPath": "/var/aos/.usrpin.txt"`
`PIN=$(cat /var/.usrpin.txt)`

**Comment by Eabramov84** [comment]
> May be I'm wrong, but please check that these two files are different and this is not misprint:
> `"userPinPath": "/var/aos/.usrpin.txt"`
> `PIN=$(cat /var/.usrpin.txt)`

Thanks for finding. fixed

**Pull request merged**